### ISSUE TITLE
salt.pillar.file_tree should support files for all hosts

### ISSUE THREAD

**Issue by dr4Ke** [saltstack/salt#32978]
### Description of Issue/Question
The only way to provide files for all hosts, in the current implementation, is to create a nodegroup containing all hosts. Then create a `nodegroups/all_hosts` directory structure in the file_tree external pillar.
This is an issue when using `salt-ssh` as it doesn't support nodegroups (nor grains) yet (#16068, #27842). And I guess it's the same with a standalone installation.

This behaviour is documented in https://docs.saltstack.com/en/latest/ref/pillar/all/salt.pillar.file_tree.html as "assigning Pillar Data to (Individual Hosts|Entire Nodegroups) but it's not clear that you can't just put your files in the root and let them be available to all nodes.

I suggest we add this ability in one of two methods:
- reference files inside the root dir and outside the `nodegroups` or `hosts` subdirectory for acces by all nodes
- add a subdirectory named `all` (for example) where we can place files for all hosts.

This way, this feature (file_tree external pillar) would be available for all flavor of Salt, even if in a limited way (no nodegroups support) for salt-ssh or standalone mode.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

- master.conf:
```
ext_pillar:
  - file_tree:
      root_dir: /srv/pillar/files
      follow_dir_links: False
      keep_newline: True
```
- /srv/pillar/files:
```
{files}
all/ {files}
hosts/host1/ {files}
hosts/host2/ {files}
nodegroups/group1/ {files}
nodegroups/group2/ {files}
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
Reading `salt/pillar/file_tree.py` `ext_pillar` function from salt version 2015.8.8 or develop.

**Comment by jfindlay** [comment]
@dr4Ke, thanks for reporting.

**Comment by The-Loeki** [comment]
Related/for cached minions solved: #39050

**Comment by lichtamberg** [comment]
Yeah, pleaseeeee add a folder for all hosts... I would really need this..

**Comment by davidjb** [comment]
Still an issue (reviving so stale bot doesn't close it)

**Comment by rallytime** [comment]
@saltstack/team-triage FYI

**Comment by davidjb** [comment]
Still an issue (reviving so stale bot doesn't close it)
